unified-codes/libs/ui/src/lib/components/entity/molecules/EntityTable.tsx

```diff
@@ -2,78 +2,51 @@ import * as React from 'react';
 
 import {
   Table,
-  TableProps,
   TableHead,
-  TableHeadProps,
   TableBody,
-  TableBodyProps,
 } from '../../data';
 
-import { IEntity, IExplorerVariables } from '@unified-codes/data';
+import { Grid } from '../../layout/atoms';
 
-import EntityTableHeader, { EntityTableHeaderProps } from './EntityTableHeader';
-import EntityTableRow, { EntityTableRowProps } from './EntityTableRow';
 export interface EntityTableProps {
-  bodyProps?: TableBodyProps;
-  data: Array<IEntity>;
-  headProps?: TableHeadProps;
-  headerProps?: EntityTableHeaderProps;
-  rowProps?: EntityTableRowProps;
-  tableProps?: TableProps;
-  variables?: IExplorerVariables;
-  onSort?: (value: string) => void;
+  classes?: {
+    body?: string,
+    head?: string,
+    paginationContainer?: string,
+    root?: string,
+    tableContainer?: string
+  };
+  header: React.ReactElement,
+  pagination: React.ReactElement,
+  rows: React.ReactElement,
 }
 
 export type EntityTable = React.FunctionComponent<EntityTableProps>;
 
 export const EntityTable: EntityTable = ({
-  tableProps,
-  headProps,
-  headerProps,
-  bodyProps,
-  rowProps,
-  data,
-  variables,
-  onSort,
+  classes,
+  header,
+  pagination,
+  rows,
 }: EntityTableProps) => {
-  const mapEntity = (entity: IEntity, index: number) => {
-    const localRowProps =
-      tableProps?.stripedRows && tableProps?.alternatingRowColour
-        ? { style: { backgroundColor: index % 2 ? tableProps?.alternatingRowColour : '' } }
-        : undefined;
-
-    return (
-      <EntityTableRow
-        key={entity.code}
-        {...{
-          ...rowProps,
-          entity,
-          rowProps: { ...localRowProps },
-        }}
-      ></EntityTableRow>
-    );
-  };
-
-  const EntityTableRows = React.useCallback(
-    () => <React.Fragment>{data.map(mapEntity)}</React.Fragment>,
-    [rowProps, data]
-  );
-  const { stripedRows, alternatingRowColour, ...otherProps } = tableProps || {};
+ 
   return (
-    <Table {...otherProps}>
-      <TableHead {...headProps}>
-        <EntityTableHeader
-          {...headerProps}
-          onSort={onSort}
-          orderDesc={variables?.orderDesc}
-          orderBy={variables?.orderBy}
-        />
-      </TableHead>
-      <TableBody {...bodyProps}>
-        <EntityTableRows />
-      </TableBody>
-    </Table>
+    <Grid classes={{ root: classes?.root }}>
+      <Grid classes={{ root: classes?.tableContainer }}>
+        <Table classes={{ root: classes?.root }}>
```

Comment:
yeah, would be nice if there was a neater way to handle the classes. This reassigning of names feels clumsy.

Reply:
Hmm, it can be made a bit tidier by extracting them into constants etc., but I think it's alright as long as the parent class fields are named appropriately to indicate the child? I got the idea from the material UI patterns for customising components: https://material-ui.com/customization/components/.